hevm.cabal

```diff
@@ -103,8 +104,17 @@ library
     Paths_hevm
   autogen-modules:
     Paths_hevm
-  if os(linux) || os(windows)
+  if impl(ghc >= 9.4) && !os(darwin)
+    -- darwin is skipped because it produces this error when building
+    -- > ghc: loadArchive: Neither an archive, nor a fat archive: `/nix/store/l3lkdfm7sg1wwc850451cikqds766h15-clang-wrapper-11.1.0/bin/clang++'
```

Comment:
I've seen this error before, iirc I was able to fix by adding the config to the executables instead of the library (that's why we have `extra-libaries: c++` duplicated a few times). Maybe that would work too for the `system-cxx-std-lib` stuff too?

Reply:
Unfortunately it does not 😞 if I apply this example diff on top of this PR, it stops building correctly. That's why I opted to generalize it for `!darwin` with the new `build-depends: system-cxx-std-lib` but keep the duplicated `extra-libraries: c++` in the executables for `darwin`

```diff
diff --git a/hevm.cabal b/hevm.cabal
index 2ebc62b6..b66c901d 100644
--- a/hevm.cabal
+++ b/hevm.cabal
@@ -196,7 +196,7 @@ executable hevm
   other-modules:
     Paths_hevm
   if os(darwin)
-    extra-libraries: c++
+    build-depends: system-cxx-std-lib
   build-depends:
     QuickCheck,
     aeson,
```

```
       > [1 of 2] Compiling Paths_hevm       ( dist/build/hevm/autogen/Paths_hevm.hs, dist/build/hevm/hevm-tmp/Paths_hevm.o, dist/build/hevm/hevm-tmp/Paths_hevm.dyn_o )
       > [2 of 2] Compiling Main             ( cli/cli.hs, dist/build/hevm/hevm-tmp/Main.o, dist/build/hevm/hevm-tmp/Main.dyn_o )
       > ghc: loadArchive: Neither an archive, nor a fat archive: `/nix/store/l3lkdfm7sg1wwc850451cikqds766h15-clang-wrapper-11.1.0/bin/clang++'
       >
       > <no location info>: error:
       >     loadArchive "/nix/store/l3lkdfm7sg1wwc850451cikqds766h15-clang-wrapper-11.1.0/bin/clang++": failed
       > /nix/store/1k85b7faqm8pc0k3y553zr8bfhb2dv2z-ghc-9.4.6/bin/ghc returned
       > ExitFailure 1
```